### PR TITLE
allow tooltips to work via clicks.

### PR DESCRIPTION
**Pull request by kbullaughey**
This PR adds a configuration option `tooltipAction` to allow tooltips to optionally work via clicks, which is better for mobile. The default value, `hover` results in the existing behavior using `mouseover` and `mouseout` but if `tooltipAction` is instead set to `click` then `click` handlers are used to show and hide tooltips. 

Allowing tooltips to work via clicks is important for supporting mobile devices which don't have `mouseover` and `mouseout` events. 

In click mode it works as follows:
1. Clicking on a cell when no tooltip is showing will show that cell's tooltip.
2. Clicking on a cell when a different cell's tooltip is showing will switch to tooltip to the newly clicked cell.
3. Clicking a cell for which a tooltip is already showing will hide that cell's tooltip.
4. Clicking anywhere else (i.e., not on the calendar plot), will hide a tooltip if one is showing.

Let me know if you have qualms with the implementation or suggestions. This is my first contrib to this project, so I'm not that familiar with the code base yet.

### CODE REVIEW

**Comment by kamisama** [comment]
Thanks for the PR.

Can you submit the PR without `grunt build`, so that only changes to src/cal-heatmap.js appears ? We use `grunt build` only when releasing a new version

File: cal-heatmap.js [hunk 1]
```diff
@@ -2503,18 +2593,28 @@ CalHeatMap.prototype = {
 				_callback({});
 				return true;
 			} else {
+				var url = this.parseURI(source, startDate, endDate);
```
Comment (kamisama) [inline cal-heatmap.js]: I think these changes are unrelated to the tooltip feature, and deserve a separate pull request

**Comment by wa0x6e** [comment]
I think you committed the last changes on the wrong branch, as they're not applied to this PR

**Comment by wa0x6e** [comment]
I'm closing this one, as it's not relevant anymore nowadays.
Changes are interesting, but we're now in 2022, and you can tap on a cell to trigger the hover event on most mobile browsers.

The only feature not supported by default is #3 (Clicking a cell for which a tooltip is already showing will hide that cell's tooltip), but we can live without.
